vm_exec.h

```diff
@@ -57,6 +57,60 @@ error !
 #define START_OF_ORIGINAL_INSN(x) /* ignore */
 #define DISPATCH_ORIGINAL_INSN(x) return LABEL(x)(ec, reg_cfp);
 
+/************************************************/
+#elif OPT_TAILCALL_THREADED_CODE
+
+// https://blog.reverberate.org/2021/04/21/musttail-efficient-interpreters.html
+// https://sillycross.github.io/2022/11/22/2022-11-22/
+// https://github.com/wasm3/wasm3/blob/main/docs/Interpreter.md#m3-massey-meta-machine
+
+// TODO: move elsewhere
+#define MUSTTAIL __attribute__((musttail))
```

Comment:
TODO: Apparently `[[clang::musttail]]` has better support (works on GCC 15)

Reply:
Hmm. I didn't actually find evidence of this, 15.1 support __attribute__ https://godbolt.org/z/3dzzGc17e so maybe this was fixed by release